zubhub_backend/zubhub/projects/views.py

```diff
@@ -586,3 +591,63 @@ def delete(self, request, *args, **kwargs):
                 profile.save()
 
             return result
+
+
+class ViolationReasonsListAPIView(ListAPIView):
+    """
+    List all violation reasons.
+
+    Requires authentication and staff or moderator.
+    Returns violations reasons.
+    """
+
+    queryset = ViolationReason.objects.all()
+    serializer_class = ViolationReasonSerializer
+    permission_classes = [IsAuthenticated, IsStaffOrModerator]
+    throttle_classes = [CustomUserRateThrottle, SustainedRateThrottle]
+
+
+class ProjectViolationsClearApiView(APIView):
+    """
+    Clear all violations for a project
+
+    Requires authentication and owner.
+    Returns project details.
+    """
+
+    permission_classes = [IsAuthenticated, IsOwner]
+    throttle_classes = [CustomUserRateThrottle, SustainedRateThrottle]
+
+    def post(self, request, *, pk):
+        try:
+            old = Project.objects.get(pk=pk)
+        except Project.DoesNotExist:
+            return Response('Not found', status=404)
+
+        if not request.user or old.creator.id != request.user.id:
+            return Response('Error', status=403)
+
+        old.violations.set([])
+        old.save()
```

Comment:
Couldn't we add this functionality to the update function in the `ProjectSerializer`? This would also prevent the user from having to make two api calls on the frontend when updating a project.

Reply:
We could, but I feel like that's going to be replaced pretty quickly in the future when actual moderation is implemented. This system is more extensible imo. I feel like 2 API calls isn't much of a problem, but I can replace this if we're sure.